justfile

```diff
@@ -0,0 +1,128 @@
+# See https://just.systems/man/en
+
+#
+# Configuration
+#
+
+export RUST_BACKTRACE := env_var_or_default("RUST_BACKTRACE", "short")
+
+# By default we compile in development mode mode because it's faster.
+build_type := if env_var_or_default("CARGO_RELEASE", "") == "" { "debug" } else { "release" }
+
+toolchain := ""
+_cargo := env_var_or_default("CARGO", "cargo" + if toolchain != "" { " +" + toolchain } else { "" })
+
+# The version name to use for packages.
+package_version := env_var_or_default("PACKAGE_VERSION", `git rev-parse --short HEAD`)
+
+# The architecture name to use for packages. Either 'amd64', 'arm64', or 'arm'.
+package_arch := env_var_or_default("ARCH", "amd64")
+
+# If a `package_arch` is specified, then we change the default cargo `--target`
+# to support cross-compilation. Otherwise, we use `rustup` to find the default.
+_cargo_target := if package_arch == "arm64" {
+        "aarch64-unknown-linux-gnu"
+    } else if package_arch == "arm" {
+        "armv7-unknown-linux-gnueabihf"
+    } else {
+        `rustup show | sed -n 's/^Default host: \(.*\)/\1/p'`
+    }
+
+# Support cross-compilation when `package_arch` changes.
+export CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER := "aarch64-linux-gnu-gcc"
+export CARGO_TARGET_ARMV7_UNKNOWN_LINUX_GNUEABIHF_LINKER := "arm-linux-gnueabihf-gcc"
+_strip := if package_arch == "arm64" { "aarch64-linux-gnu-strip" } else if package_arch == "arm" { "arm-linux-gnueabihf-strip" } else { "strip" }
+
+_target_dir := join("target", _cargo_target, build_type)
+_target_bin := join(_target_dir, "linkerd-await")
+_package_name := "linkerd-await-" + package_version + "-" + package_arch
+_package_dir := join("target/package", _package_name)
+_shasum := "shasum -a 256"
+
+# If we're running in Github Actions and cargo-action-fmt is installed, then add
+# a command suffix that formats errors.
+_fmt := if env_var_or_default("GITHUB_ACTIONS", "") != "true" { "" } else {
+    ```
+    if command -v cargo-action-fmt >/dev/null 2>&1; then
+        echo "--message-format=json | cargo-action-fmt"
+    fi
+    ```
+}
+
+# Use nextest if it's available (and we're not in CI). Mostly to work around
+# https://github.com/nextest-rs/nextest/issues/422
```

Comment:
fwiw, i think the upstream issue has been resolved (https://github.com/nextest-rs/nextest/blob/main/site/src/CHANGELOG.md#0932---2022-07-30) so we may want to switch back to running nextest on CI (if it's any faster for us?)